sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -844,6 +850,35 @@ Exceptions:
 * Throws synchronously with error code `invalid` if the topology of `graph` is
   not the same as the existing graph topology, or if the nodes were not added in
   the same order.
+
+  |
+[source, c++]
+----
+using namespace ext::oneapi::experimental;
+template <typename T>
+void command_graph<graph_state::executable> 
+update_parameter(const T& from, const T& to);
+----
+
+|Updates a parameter across the executable graph such that all nodes in the
+graph which use `from` as a parameter will have that parameter replaced by
+`to`.
+
+Preconditions:
+
+* This member function is only available when the `command_graph` state is
+  `graph_state::executable`.
+* T must be either a buffer or a raw pointer to a USM allocation.
```

Comment:
If this restriction is going to stay, we might want to rethink the function name `update_parameter<T>`. Which is kind of generic, and might lead to confusion. To update for example a scalar value we would have to fall back to the `update` function, correct?

Reply:
Pushed a change which renames the entry-point to `update_memory_parameter<T>` and mentions the scalar limitation, since I believe it will stay.

I think the `update()` function should be able to support scalars because you can set the scalar argument from the argument at the same index in the passed modifiable graph, and it doesn't matter if that's a copy of the original user value. Maybe once we start implementing that we'll discover that there is a detail that prevents using scalars with `update`, but my hope would be it is possible.

Reply:
Scalars should be fine I believe, anything that is set as a kernel argument should be really. I think conceptually they're a bit more difficult to include in the spec because they're not really a formal concept in SYCL, rather they are just things which happen to be captured by lambdas or functors that are set as kernel arguments in the runtime out of the view of the user. But that shouldn't prevent us from doing it, just makes it a bit trickier I think.

Reply:
After some more thought I think that actually scalars are a bit problematic. Much like buffers we quickly lose visibility of the actual type of the scalar argument, instead we just have a bunch of bytes and the size of the data. This makes comparing the `from` parameter difficult because if we're just comparing bytes there is a potential for collisions where you could pass an incorrect `from` but the bytes match so it is accepted.